src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -126,20 +127,34 @@ public async Task<(Error, GitHubUser)> GetByCommit(string authorEmail, string re
             return await GetByLogin(login);
         }
 
-        public Task SaveChanges()
+        public async Task SaveChanges(Config config)
         {
             if (!_updated)
             {
-                return Task.CompletedTask;
+                return;
             }
 
+            string file;
             lock (_lock)
             {
-                var file = new GitHubUserCacheFile
+                file = JsonUtility.Serialize(new GitHubUserCacheFile
                 {
                     Users = Users.ToArray(),
-                };
-                return ProcessUtility.WriteFile(_cachePath, JsonUtility.Serialize(file));
+                });
+            }
+            await ProcessUtility.WriteFile(_cachePath, file);
+            var url = config.GitHub.UserCache;
+            if (config.GitHub.UpdateRemoteUserCache && HrefUtility.IsHttpHref(url))
+            {
+                try
+                {
+                    // TOOD: aware of ETag to handle conflicts
+                    await HttpClientUtility.PutAsync(config.GitHub.UserCache, new StringContent(file), config);
+                }
+                catch (HttpRequestException ex)
+                {
+                    throw Errors.UploadFailed(url, ex.Message).ToException();
```

Comment:
Do we want to crush the entire build here if fail to update remote cache?

Reply:
Good question... `download-failed` is an error, but seems `upload-failed` doesn't as severe as it as it does not affect the main output.